src/vs/workbench/contrib/debug/test/node/streamDebugAdapter.test.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as net from 'net';
+import * as platform from 'vs/base/common/platform';
+import { tmpdir } from 'os';
+import { join } from 'vs/base/common/path';
+import { SocketDebugAdapter, NamedPipeDebugAdapter, StreamDebugAdapter } from 'vs/workbench/contrib/debug/node/debugAdapter';
+
+function rndPort(): number {
+	const min = 8000;
+	const max = 9000;
+	return Math.floor(Math.random() * (max - min) + min);
+}
+
+function rndName(): string {
```

Comment:
small tip: I usually just use crypto.randomBytes in cases like these. Slower but works just fine for tests.

Reply:
Done.